src/test/java/uk/gov/hmcts/reform/pdf/service/client/PDFServiceClientTest.java

```diff
@@ -0,0 +1,94 @@
+    private PDFServiceClient pdfServiceClient;
+
+    @Before
+    public void setup() {
+        pdfServiceClient = new PDFServiceClient(restClient, URI.create(endpointBase));
+    }
+
+    @Test
+    public void requests_new_endpoint() {
+        pdfServiceClient.generateFromHtml(sampleTemplate.getBytes(), new HashMap<>());
+
+        ArgumentCaptor<URI> uriArgumentCaptor = ArgumentCaptor.forClass(URI.class);
+        verify(restClient).postForObject(uriArgumentCaptor.capture(), any(), any());
+        assertThat(uriArgumentCaptor.getValue().toString()).isEqualTo(endpointBase + "/pdfs");
+    }
+
+    @Test
+    public void request_content_type_contains_versioned_mime_type() {
+        pdfServiceClient.generateFromHtml(sampleTemplate.getBytes(), new HashMap<>());
+
+        ArgumentCaptor<HttpEntity> httpEntityArgumentCaptor = ArgumentCaptor.forClass(HttpEntity.class);
+        verify(restClient).postForObject(any(), httpEntityArgumentCaptor.capture(), any());
+
+        assertThat(httpEntityArgumentCaptor.getValue().getHeaders().getContentType())
+            .isEqualTo(PDFServiceClient.API_VERSION);
+    }
+
+    @Test
+    public void request_accepts_pdf() {
+        ArgumentCaptor<HttpEntity> httpEntityArgumentCaptor = ArgumentCaptor.forClass(HttpEntity.class);
+
+        pdfServiceClient.generateFromHtml(sampleTemplate.getBytes(), new HashMap<>());
+
+        verify(restClient).postForObject(any(), httpEntityArgumentCaptor.capture(), any());
+
+        assertThat(httpEntityArgumentCaptor.getValue().getHeaders().getAccept()).contains(MediaType.APPLICATION_PDF);
+    }
+
+    @Test
+    public void template_contents_are_sent_as_plain_string() throws IOException {
+        ArgumentCaptor<HttpEntity> httpEntityArgumentCaptor = ArgumentCaptor.forClass(HttpEntity.class);
+
+        pdfServiceClient.generateFromHtml(sampleTemplate.getBytes(), new HashMap<>());
+
+        verify(restClient).postForObject(any(), httpEntityArgumentCaptor.capture(), any());
+
+        GeneratePdfRequest generatePdfRequest = new ObjectMapper()
+            .readValue(httpEntityArgumentCaptor.getValue().getBody().toString(), GeneratePdfRequest.class);
+
+        assertThat(generatePdfRequest.template).isEqualTo(sampleTemplate);
+    }
+
+    @Test
+    public void values_are_passed_along() throws IOException {
+        Map<String, Object> values = new HashMap<>();
+        values.put("hello", "World!");
+        values.put("Foo", "bar");
+
+        pdfServiceClient.generateFromHtml(sampleTemplate.getBytes(), values);
+
+        ArgumentCaptor<HttpEntity> httpEntityArgumentCaptor = ArgumentCaptor.forClass(HttpEntity.class);
+        verify(restClient).postForObject(any(), httpEntityArgumentCaptor.capture(), any());
+
+        GeneratePdfRequest generatePdfRequest = new ObjectMapper()
+            .readValue(httpEntityArgumentCaptor.getValue().getBody().toString(), GeneratePdfRequest.class);
+
+        assertThat(generatePdfRequest.values).containsAllEntriesOf(values);
```

Comment:
I would narrow this assertion down to check that only expected values are passed in (and update test name accordingly).

Reply:
What would it look like?

Reply:
You can use `containsExactly` or add an assertion on the size of that collection.

Reply:
But all that matters is if given values are passed along, regardless of their order. It also doesn't matter if client added anything as long as original values are there.

Reply:
You are probably correct, but having additional values popping in there is an unexpected side effect. Changing the assertion is easy and cheap to do, I think it's worth updating just to be on the safe side.

Reply:
done